src/app/TideSearchApplication.cpp

```diff
@@ -1868,44 +1914,32 @@ void TideSearchApplication::calcResidueScoreCount (
   //set to zero now that score counts for first amino acid are in matrix
   dynProgArray[initCountRow][initCountCol] = 0.0;
 
-  // populate matrix with score counts for non-terminal amino acids in sequence
-  for (ma = colFirst; ma < colLast; ma++) {
-    col = maxAaMass + ma;
-
+//The following code was added by AKF to make the DP calculation faster
+  int newCol;
+  for (ma = initCountCol+1; ma < colLast; ++ma) {
+    col = ma;  
     for (de = 0; de < nAa; de++) {
-      aaMassCol[de] = col - aaMass[de];
+      aaMassCol[de] = col + aaMass[de];
     }
-    for (row = rowFirst; row <= rowLast; row++) {
-      sumScore = dynProgArray[row][col];
-      for (de = 0; de < nAa; de++) {
-        evidRow = row - residueEvidenceMatrix[de][ma];
-        //sumScore += dynProgArray[ evidRow ][ aaMassCol[ de ] ];
-        sumScore += dynProgArray[evidRow][aaMassCol[de]] * aaFreqI[de];
+    for (row = rowFirst; row <= rowLast; ++row) {
+      if (dynProgArray[row][col] == 0.0) {
+        continue;
       }
-      dynProgArray[row][col] = sumScore;
-    }
-  }
-
-  // populate matrix with score counts for last (i.e. C-terminal) amino acid in sequence
-  ma = colLast;
-  col = maxAaMass + ma;
-
-  //no evidence should be added for last amino acid in sequence
-  evid = 0;
-  for (de = 0; de < nAa; de++) {
-    aaMassCol[de] = col - aaMass[de];
-  }
-  for (row = rowFirst; row <= rowLast; row++) {
-    evidRow = row - evid;
-    sumScore = 0.0;
-    for (de = 0; de < nAa; de++) {
-      //sumScore += dynProgArray[ evidRow ][ aaMassCol[ de ] ];
-      sumScore += dynProgArray[evidRow][aaMassCol[de]] * aaFreqC[de];
-    }
-    dynProgArray[row][col] = sumScore;
-  }
+      for (de = 0; de < nAa; de++) {
+        newCol = aaMassCol[de];
+        if (newCol == colLast) {
```

Comment:
Wouldn't it be slightly faster to put the following two if statements into an if + else if?

if (newCol < colLast) { } else if (newCol == colLast)

Reply:
Correct. I fixed it.